dali/pipeline/data/tensor_vector.cc

```diff
@@ -138,18 +136,22 @@ void TensorVector<Backend>::set_type(const TypeInfo &new_type) {
   }
 }
 
+
 template <typename Backend>
 const TypeInfo &TensorVector<Backend>::type() const {
   if (state_ == State::contiguous) {
     return tl_->type();
   }
-  if (tensors_.size() > 0) {
-    return tensors_[0]->type();
+  if (tensors_.size() == 0) {
+    return type_;
   }
-  return type_;
+  for (size_t i = 1; i < tensors_.size(); i++) {
+    assert(tensors_[0]->type() == tensors_[i]->type());
+  }
```

Comment:
I think it's a bit excessive to test it in every access to the type.

Reply:
But it's only in debug conf, right? Cause compiler with optimize out this for loop in release. In this case, I think it'd be nice to have this extra sanity check